src/util/mongo.controller.js

```diff
@@ -62,6 +62,29 @@ const updateResource = async (id, data, resourceType) => {
   return { id: results.value.id, created: false };
 };
 
+/**
+ * searches for a document and updates it by pushing to existing if found, creates it if not
+ * @param {*} id id of resource to be updated
+ * @param {*} data the new data to push in the document
+ * @param {*} resourceType the collection the document is in
+ * @returns the id of the updated/created document
+ */
+const pushToResource = async (id, data, resourceType) => {
+  const collection = db.collection(resourceType);
+
+  const results = await collection.findOneAndUpdate({ id: id }, { $push: data });
```

Comment:
pushing multiple requires an $each i.e. data = { scores: { $each: [ 90, 92, 85 ] } }
should $each be passed in as data, or is that bad modularization? (currently being passed in as data)

Reply:
I think the way you've structured this is fine. We have other functions in the mongo controller that accept arbitrary mongo queries as arguments and it's been pretty flexible and beneficial, so I'm fine with leaving the arguments/function call how you've written it here